lib/msal-common/src/authority/Authority.ts

```diff
@@ -579,47 +580,38 @@ export class Authority {
             this.correlationId
         );
 
-        this.performanceClient?.setPreQueueTime(
-            PerformanceEvents.RegionDiscoveryDetectRegion,
-            this.correlationId
-        );
-        const autodetectedRegionName = await this.regionDiscovery.detectRegion(
-            this.authorityOptions.azureRegionConfiguration?.environmentRegion,
-            this.regionDiscoveryMetadata
-        );
-
-        const azureRegion =
-            this.authorityOptions.azureRegionConfiguration?.azureRegion ===
-            Constants.AZURE_REGION_AUTO_DISCOVER_FLAG
-                ? autodetectedRegionName
-                : this.authorityOptions.azureRegionConfiguration?.azureRegion;
+        const userConfiguredAzureRegion = this.authorityOptions.azureRegionConfiguration?.azureRegion;
 
-        if (
-            this.authorityOptions.azureRegionConfiguration?.azureRegion ===
-            Constants.AZURE_REGION_AUTO_DISCOVER_FLAG
-        ) {
-            this.regionDiscoveryMetadata.region_outcome = autodetectedRegionName
-                ? RegionDiscoveryOutcomes.AUTO_DETECTION_REQUESTED_SUCCESSFUL
-                : RegionDiscoveryOutcomes.AUTO_DETECTION_REQUESTED_FAILED;
-        } else {
-            if (autodetectedRegionName) {
-                this.regionDiscoveryMetadata.region_outcome =
-                    this.authorityOptions.azureRegionConfiguration
-                        ?.azureRegion === autodetectedRegionName
-                        ? RegionDiscoveryOutcomes.CONFIGURED_MATCHES_DETECTED
-                        : RegionDiscoveryOutcomes.CONFIGURED_NOT_DETECTED;
-            } else {
-                this.regionDiscoveryMetadata.region_outcome =
-                    RegionDiscoveryOutcomes.CONFIGURED_NO_AUTO_DETECTION;
+        if (userConfiguredAzureRegion) {
+            if (userConfiguredAzureRegion !== Constants.AZURE_REGION_AUTO_DISCOVER_FLAG) {
+                this.regionDiscoveryMetadata.region_outcome = RegionDiscoveryOutcomes.CONFIGURED_NO_AUTO_DETECTION;
+                this.regionDiscoveryMetadata.region_used = userConfiguredAzureRegion;
```

Comment:
Do we cover for user passing an `invalid region`? How does the e2e flow work there? Does the service throw?